compiler-rt/lib/sanitizer_common/sanitizer_file.cpp

```diff
@@ -104,6 +111,16 @@ void ReportFile::SetReportPath(const char *path) {
     }
   }
 
+#ifdef HAS_GETAUXVAL
+  if (getauxval(AT_SECURE) != 0 && path &&
+      internal_strcmp(path, "stderr") != 0 &&
+      internal_strcmp(path, "stdout") != 0) {
+    Report(
+        "ERROR: Permission denied setting log_path for a binary in a secure context. You must run on a same priviledge level\n");
```

Comment:
nit: typo in "privilege".

Also I know what this is fixing, but I still don't really understand "You must run on a same privilege level". Also I feel talking about permission denied implies something is wrong with the filesystem permissions of the given file.

Maybe something like: "log_path must be 'stderr' or 'stdin' for AT_SECURE (e.g. setuid binaries), is '%s'"

Reply:
Actually couldn't agree more :)